kolibri/plugins/user/assets/src/views/SignUpPage.vue

```diff
@@ -189,6 +190,10 @@
         // query is before hash
         return getUrlParameter('next');
       },
+      showPasswordInput() {
+        this.setEmptyPassword();
```

Comment:
`computed` props are meant to be read-only. It's not good to call a `method` which modifies `data` from within one.

A more conventional strategy would be to define a new computed prop like:

```
passwordToSave() {
  if (this.facilityConfig.learner_can_login_with_no_password && this.password === '') {
    return 'NOT_SPECIFIED'
  }
  return this.password;
},
```

That said, I don't actually think the front-end should be setting the `NOT_SPECIFIED` string at all. I would expect this to be handled completely by the backend, because the string needs to bypass Django's normal hashing and salting mechanisms.

Reply:
> `computed` props are meant to be read-only. It's not good to call a `method` which modifies `data` from within one.
> 
> A more conventional strategy would be to define a new computed prop like:
> 
> ```js
> passwordToSave() {
>   if (this.facilityConfig.learner_can_login_with_no_password && this.password === '') {
>     return 'NOT_SPECIFIED'
>   }
>   return this.password;
> },
> ```
> 

Totally agree, thank you. I will change it.


> That said, I don't actually think the front-end should be setting the `NOT_SPECIFIED` string at all. I would expect this to be handled completely by the backend, because the string needs to bypass Django's normal hashing and salting mechanisms.

The front-end must send something, either'`NOT_SPECIFIED`, an empty string, or another keyword. It's a question of deciding what we want to do, and it's an open question I left in the PR description to be decided between the people involved in this. We need a convention for the backend to know that this is an user without password.
Said that: the backend will work fine in the part where it  manages the self-creation of the user. However, when handling the form where the admin creates users there's a pre-validation in the backend that requires a non-empty password. Code will be simpler and will require less changes using any keyword like `NOT_SPECIFIED` than sending an empty string.